codebase/general/src.lib/dmap.1.25/include/dmap.h

```diff
@@ -34,6 +34,8 @@
 #ifndef _DMAP_H
 #define _DMAP_H
 
+#include <zlib.h>
```

Comment:
Why is it necessary to include `<zlib.h>` in this header file?

Reply:
I am not 100% how RST's build script works regarding linking as it seems to be a bit of a mess. But if I remove the zlib header from that file I get errors:
`rang_badlags.c:87:10: warning: variable ‘nave’ set but not used [-Wunused-but-set-variable]
   double nave;
          ^~~~
cc -fPIC -Wall -O3 -ansi -D_GNU_SOURCE -D_LINUX -I/home/marina/superdarn/rst//include/base -I/home/marina/superdarn/rst//include/general -I/home/marina/superdarn/rst//include/superdarn -c noise_stat.c    
cc -fPIC -Wall -O3 -ansi -D_GNU_SOURCE -D_LINUX -I/home/marina/superdarn/rst//include/base -I/home/marina/superdarn/rst//include/general -I/home/marina/superdarn/rst//include/superdarn -c fit_acf.c    
fit_acf.c: In function ‘fit_acf’:
fit_acf.c:57:42: warning: variable ‘sum_kphi’ set but not used [-Wunused-but-set-variable]
             sum_p,sum_pk,sum_pk2,sum_phi,sum_kphi, t0,t2,t4,*phi_res=NULL;
                                          ^~~~~~~~
fit_acf.c:57:34: warning: variable ‘sum_phi’ set but not used [-Wunused-but-set-variable]
             sum_p,sum_pk,sum_pk2,sum_phi,sum_kphi, t0,t2,t4,*phi_res=NULL;
                                  ^~~~~~~
cc -fPIC -Wall -O3 -ansi -D_GNU_SOURCE -D_LINUX -I/home/marina/superdarn/rst//include/base -I/home/marina/superdarn/rst//include/general -I/home/marina/superdarn/rst//include/superdarn -c do_fit.c    
In file included from /home/marina/superdarn/rst//include/superdarn/rprm.h:36:0,
                 from /home/marina/superdarn/rst//include/superdarn/fitacf.h:12,
                 from do_fit.c:45:
/home/marina/superdarn/rst//include/general/dmap.h:63:5: error: unknown type name ‘gzFile’
     gzFile z;
     ^~~~~~
/home/marina/superdarn/rst//include/general/dmap.h:181:35: error: unknown type name ‘gzFile’
 struct DataMap *DataMapReadBlockZ(gzFile file,int *s);
                                   ^~~~~~
/home/marina/superdarn/rst//include/general/dmap.h:182:30: error: unknown type name ‘gzFile’
 struct DataMap *DataMapReadZ(gzFile file);
                              ^~~~~~
/home/marina/superdarn/rst//include/general/dmap.h:183:19: error: unknown type name ‘gzFile’
 int DataMapWriteZ(gzFile file,struct DataMap *ptr);
                   ^~~~~~
make: *** [/home/marina/superdarn/rst//build/make/makelib.linux:19: do_fit.o] Error 1
Compilation Aborted.
`

In the header files (I changed) itself you call the data type there like: 'gzFile’ so unknown type errors were an issue. 

So I took the path of least resistance and added that header file in. I could leave a comment about it if you want?

Reply:
If I understand correctly, this largely breaks from past RST practices, where `#includes` are usually handled in the individual `.c` files rather than in header files.  At a minimum I think an `#ifndef` should be used beforehand if this is deemed absolutely necessary.

Reply:
I will address this right away. Maybe in the future, we could look into actually making an RST practice manual :) However, there is no best practice with header location. Typically you want to keep them in the `.c` file but sometimes need them in `.h` file if you are referencing a structure or else you getting some warnings or errors. 

But yes, I do need to add those `#ifndef` :)

Reply:
done